### PR TITLE
Consolidate information on running the testsuite in a README

### PR DESCRIPTION
**Pull request by wilzbach**
So that hopefully people in the future have it easy to get started.

From: https://github.com/dlang/dmd/pull/8038#issuecomment-373524698

> Might be nice to add a README.md to this directory. That explains what variables are available to BASH tests. Maybe move the documentation from the Makefile to the README file and have it explain everything there.

CC @marler8997

### CODE REVIEW

File: test/Makefile [hunk 1]
```diff
@@ -1,90 +1,3 @@
...
-#
-#    run_tests:          run all tests
-#    run_runnable_tests:         run just the runnable tests
-#    run_compilable_tests:       run just the compilable tests
-#    run_fail_compilation_tests: run just the fail compilation tests
-#
-#    quick:              run all tests with no default permuted args
-#                        (individual test specified options still honored)
-#
-#    clean:              remove all temporary or result files from prevous runs
```
Comment (wilzbach) [inline test/Makefile]: Typo: prevous

File: test/Makefile [hunk 2]
```diff
@@ -1,90 +1,3 @@
...
-#
-#    test_results/compilable/json.d.out      runs an individual test
-#                                            (run log of the test is stored)
-
-# In-test variables
-# -----------------
-#
-#   COMPILE_SEPARATELY:  if present, forces each .d file to compile separately and linked
-#                        together in an extra setp.
```
Comment (wilzbach) [inline test/Makefile]: Typo: setp

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+-----------
+
+### Run all tests
+
+```sh
+make -j10
+```
+
+Note: the exact parallelism parameter `-j` depends on your system,
+but if you have enough memory, a high `-j` allows a significant speed-up.
```
Comment (wilzbach) [inline test/README.md]: FWIW on my machine i7 4th Gen with 16GB RAM I usually get the best results with `-j20`.
Running all `fail_compilation` tests takes merely 7s (with `-j10` it's 10s)

**Review by wilzbach — COMMENTED** [review]
(marked the typos I found and corrected)

File: test/README.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+Multiple arguments are supported too.
+You can use `./run_individual_tests` to quickly run a custom subset of tests.
+For example, all diagnostic tests in `fail_compilation`:
+
+```sh
+./run_individual_tests fail_compilation/diag*.d
+```
+
+### Automatically update the error messages
```
Comment (JinShil) [inline test/README.md]: I don't understand what this means.  What error messages is it updating and why would someone want to do that?

Comment (JinShil) [reply test/README.md]: Ah, you mean this automatically overwrites the TEST_OUTPUT for the given tests?

**Review by jacob-carlborg — APPROVED** [review]

**Comment by wilzbach** [comment]
> I don't understand what this means. What error messages is it updating and why would someone want to do that?
> Ah, you mean this automatically overwrites the TEST_OUTPUT for the given tests?

Yes, I reworded this section and hopefully made it clearer.
In case you missed it, `AUTO_UPDATE=1` has been introduced two weeks ago: https://github.com/dlang/dmd/pull/8010

While improving the AUTO_UPDATE=1 text, I also realized that there's a small bit about AUTO_UPDATE=1 which we should improve -> https://github.com/dlang/dmd/pull/8081

**Pull request merged**